media-libs/libsdl2/patches/libsdl2-2.0.20.patchset

```diff
@@ -1265,3 +1265,66 @@ index c5026de..868512e 100644
  	mouse->FreeCursor = HAIKU_FreeCursor;
 -- 
 2.30.2
+
+
+From d4462683c13c9aed712dbbe2f895992ffafd44f4 Mon Sep 17 00:00:00 2001
+From: Ivan Holmes <ivan@ivanholmes.co.uk>
+Date: Thu, 5 May 2022 22:58:29 +0100
+Subject: Don't create BApplication if one already exists
+
+
+diff --git a/src/main/haiku/SDL_BeApp.cc b/src/main/haiku/SDL_BeApp.cc
+index a5e1428..cc8e77b 100644
+--- a/src/main/haiku/SDL_BeApp.cc
++++ b/src/main/haiku/SDL_BeApp.cc
+@@ -54,27 +54,29 @@ const char *signature = "application/x-SDL-executable";
+ static int
+ StartBeApp(void *unused)
+ {
+-    BApplication *App;
+-
+-    // dig resources for correct signature
+-    image_info info;
+-    int32 cookie = 0;
+-    if (get_next_image_info(B_CURRENT_TEAM, &cookie, &info) == B_OK) {
+-        BFile f(info.name, O_RDONLY);
+-        if (f.InitCheck() == B_OK) {
+-            BAppFileInfo app_info(&f);
+-            if (app_info.InitCheck() == B_OK) {
+-                char sig[B_MIME_TYPE_LENGTH];
+-                if (app_info.GetSignature(sig) == B_OK)
+-                    signature = strndup(sig, B_MIME_TYPE_LENGTH);
++    if (!be_app) {
```

Comment:
Rather than indenting everything, just put this check at the start of the function. Or, better yet, do not invoke `StartBeApp` at all if `be_app` is already set.

Reply:
Okay, I'll keep working on it and give this a go